packages/cw1155/src/msg.rs

```diff
@@ -0,0 +1,48 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{Binary, HumanAddr, Uint128};
+
+pub type TokenId = String;
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum Cw1155HandleMsg {
+    /// TransferFrom is a base message to move tokens, if `env.sender` has sufficient pre-approval.
+    TransferFrom {
+        // `None` means minting
+        from: Option<HumanAddr>,
+        // `None` means burning
+        to: Option<HumanAddr>,
+        token_id: TokenId,
+        value: Uint128,
+    },
+    /// SendFrom is a base message to move tokens to contract
+    /// if `env.sender` has sufficient pre-approval.
```

Comment:
-> if `env.sender` is the owner or has sufficient pre-approval.

Reply:
done